process_sample_data.py

```diff
@@ -84,16 +85,31 @@ def _parseData(self):
                             .timestamp() * 1000.0)
 
             self._flow_rates.append(float(splitDatum[1]
-                                          .replace(flow_rate_marker,
-                                                   "")
+                                          .replace(flow_rate_marker, "")
                                           .strip("\n")) / 10)
             self._tidal_volumes.append(float(splitDatum[2]
-                                             .replace(tidal_volume_marker,
-                                                      "")
+                                             .replace(tidal_volume_marker, "")
                                              .strip("\n")))
 
             if (pressure_marker in datum):
                 self._pressures.append(float(splitDatum[3]
                                              .replace(pressure_marker,
                                                       "")
                                              .strip("\n")) / 10)
+
+    def _remove_repeat_data(self):
+        non_repeat_indices = []
+        for i in range(len(self._timestamps)):
+            if (not (math.isclose(self._flow_rates[i],
+                                  self._flow_rates[i-1])
+                     and math.isclose(self._tidal_volumes[i],
+                                      self._tidal_volumes[i-1])
+                     and (math.isclose(self._pressures[i],
+                                       self._pressures[i-1])
+                          if len(self._pressures) > 0 else True))):
```

Comment:
Is the length of `_pressures` ever different from the length of the other lists?

Reply:
yes, I have one data set that has flow rate, tidal volume, and pressure, and another than only has flow rate and tidal volume.

Reply:
Ok, so what's the desired behavior out past the end of the shorter of the two? It's not clear at this point.